mergify_engine/rules/filter.py

```diff
@@ -258,6 +271,13 @@ async def _op(obj: GetAttrObjectT) -> FilterResultT:
                     typing.Awaitable[typing.Any], ref_values_expanded
                 )
 
+            if self.binary_op_hook is not None:
+                res = self.binary_op_hook(
+                    binary_op, iterable_op, ref_values_expanded, attribute_name
+                )
+                if res is not None:
+                    return res
+
```

Comment:
I'm not a fan of the hook approach here. I'd rather do a classic inheritance.

Reply:
I can try, I was afraid to fight with mypy.